src/grammar.pegjs

```diff
@@ -0,0 +1,173 @@
+// Filepath Parsing Grammar
+// ==========================
+//
+// Paste into https://pegjs.org/online
+
+Path
+  = RemotePath
+  / FileSchemaPath
+  / AbsolutePath
+  / RelativePath
+
+RemotePath
+  = protocol:RemoteProtocol origin:Origin Sep root:PosixRoot path:PathWrapper {
+    return {
+      protocol,
+      implicitProtocol: false,
+      origin,
+      absolute: true,
+      root,
+      ...path
+    }
+  }
+
+  RemoteProtocol
+    = HttpProtocol
+    / HttpsProtocol
+
+    HttpProtocol
+      = raw:"http://"i {
+      return 'http'
+	}
```

Comment:
Oh god. Fricking tabs vs spaces in my indenting because of that online PEGjs editor.

Reply:
fixed